modules/dag-l0/src/main/scala/org/tessellation/dag/l0/domain/snapshot/programs/GlobalSnapshotEventCutter.scala

```diff
@@ -0,0 +1,116 @@
+package org.tessellation.dag.l0.domain.snapshot.programs
+
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.contravariantSemigroupal._
+import cats.syntax.either._
+import cats.syntax.eq._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.semigroupal._
+import cats.syntax.traverse._
+import cats.syntax.unorderedFoldable._
+
+import org.tessellation.dag.l0.infrastructure.snapshot.{DAGEvent, StateChannelEvent}
+import org.tessellation.ext.kryo._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.node.shared.domain.event.EventCutter
+import org.tessellation.schema.Block
+import org.tessellation.security.hash._
+import org.tessellation.security.{Hashed, Hasher}
+
+import eu.timepit.refined.types.numeric.PosInt
+
+object GlobalSnapshotEventCutter {
+  def make[F[_]: Async: KryoSerializer: Hasher](maxBinarySizeBytes: PosInt): EventCutter[F, StateChannelEvent, DAGEvent] =
+    new EventCutter[F, StateChannelEvent, DAGEvent] {
+
+      def cut(scEvents: List[StateChannelEvent], dagEvents: List[DAGEvent]): F[(List[StateChannelEvent], List[DAGEvent])] =
+        doesNotExceedMaxSize(scEvents, dagEvents)
+          .ifM((scEvents, dagEvents).pure[F], removeEvents(scEvents, dagEvents))
+
+      def doesNotExceedMaxSize(scEvents: List[StateChannelEvent], dagEvents: List[DAGEvent]): F[Boolean] =
+        (scEvents.toBinary.liftTo[F], dagEvents.toBinary.liftTo[F])
+          .mapN((a, b) => a.length + b.length <= maxBinarySizeBytes.value)
```

Comment:
I see that you only compare the size of `List[StateChannelEvent] + List[DAGEvent]` instead of comparing the size of the whole produced `GlobalSnapshot` proposal. Is it by design?

I am asking because `CurrencyEventsCutter` compares the whole `CurrencySnapshot` proposal by recursively recreating it until the whole proposal fits the limit (not just the blocks part of it).

Reply:
Where this is plugged in the `GlobalSnapshot` has already been separated into two parts, and some filtering of invalid values removed so I had to cutter work on those two parts. The max size is effectively arbitrary; does it matter if we're checking the size of the resulting parts rather than the original data structure?

cc: @TheMMaciek @ryle-goehausen

Reply:
I think it doesn't matter as long as we don't want the whole snapshot to not exceed some limit (and I guess it's not the case). Also if we impose validation we will need to only validate the size of the events and not a size of a whole snapshot.